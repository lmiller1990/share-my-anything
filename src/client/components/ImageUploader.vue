<template>
  <div id="upload-form">
		<form class="form-inline" enctype="multipart/form-data">
				<label class="btn btn-default btn-primary">
					Click to share yours!
					<input type="file" 
						id="image-button" name="image" 
						@change="onImageChange" hidden>
				</label>

			<input 
			  v-if="imageChosen" 
				v-model="category" 
				placeholder="category..."
				class="form-control form-control-lg" 
				name="category">

			<button 
				v-show="imageChosen && category"
				id="save-button"
				class="btn btn-default btn-lg"
				@click="save">
				Share
			</button>

		</form>
  </div>
</template>

<script>
	import 'isomorphic-fetch'
	import { createImageEndpointRoute } from '../../shared/routes'

  export default {
    name: 'ImageUploader',

		data() {
			return {
				imageChosen: false,
				formData: null,
				category: null
			}
		},

		methods: {
			onImageChange(e) {
				const fieldName = e.target.name
				const fileList = e.target.files
				const fd = new FormData()
				
				if (!fileList.length) {
					this.imageChosen = false
					return
				}
				this.imageChosen = true

				fd.append(fieldName, fileList[0], fileList[0].name)
				this.formData = fd

				return fd
			},

			save(evt) {
				evt.preventDefault()
				this.formData.append('category', this.category)

				fetch(createImageEndpointRoute(), { 
					method: 'POST', 
					body: this.formData
				})
				.then((res) => {
					if (!res.ok) throw Error('Error', res)
					return res.status
				})
				.then((data) => {
					console.log(data)
					return data
				})
			},
		}
  }
</script>

<style scoped>
#upload-form {
	padding: 7px;
}
</style>
