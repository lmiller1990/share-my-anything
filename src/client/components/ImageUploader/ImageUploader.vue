<template>
  <div>
		<div class="upload container">
			<h4 class="upload form">
				Share
			</h4>
			<form enctype="multipart/form-data">
				<input type="file" name="image" @change="onImageChange">
				<input type="button" id="save-button" @click="save" value="Submit">
			</form>
		</div>
  </div>
</template>

<script>
	import 'isomorphic-fetch'
	import { createImageEndpointRoute } from '../../../shared/routes'

  export default {
    name: 'ImageUploader',

		data() {
			return {
				imageChosen: false,
				formData: null
			}
		},

		methods: {
			onImageChange(e) {
				const fieldName = e.target.name
				const fileList = e.target.files
				const fd = new FormData()
				
				if (!fileList.length) 
					return;

				fd.append(fieldName, fileList[0], fileList[0].name)
				this.formData = fd

				return fd
			},

			save() {
				console.log('SSAVEEEE')
				fetch(createImageEndpointRoute(), { 
					method: 'POST', 
					body: this.formData
				})
				.then((res) => {
					if (!res.ok) throw Error('Error', res)
					return res.status
				})
				.then((data) => {
					console.log(data)
					return data
				})
			},
		}
  }
</script>

<style scoped>
</style>
