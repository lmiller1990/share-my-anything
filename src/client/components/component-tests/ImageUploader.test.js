import ImageUploader from '../ImageUploader.vue'
import { shallow, mount } from 'avoriaz'
import Vue from 'vue'

describe('ImageUploader.vue', () => {
	it('save should return a 200 status', () => {
		const wrapper = mount(ImageUploader, {
			propsData: {
				imageChosen: true,
				category: "something"
			}
		})
		wrapper.update()
		const saveMock = jest.fn()
		wrapper.setMethods({ save: saveMock })

		wrapper
		wrapper.find('#save-button')[0].trigger('click')
		expect(saveMock.mock.calls.length).toBe(1)
	})

	it('change should trigger onImageChange to be called', () => {
		const wrapper = shallow(ImageUploader)
		const change = jest.fn()
		wrapper.setMethods({ onImageChange: change })

		wrapper.find('#image-button')[0].trigger('change')
		expect(change.mock.calls.length).toBe(1)
	})

	it('matches the snapshot', () => {
		const wrapper = mount(ImageUploader)
		expect(wrapper.$el).toMatchSnapshot()
	})
})
