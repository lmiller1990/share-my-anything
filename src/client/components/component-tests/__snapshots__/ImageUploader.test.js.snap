// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUploader.vue matches the snapshot 1`] = `undefined`;
